tasks/post_copy_request_to_queue/test/unit_tests/test_post_copy_request_to_queue.py

```diff
@@ -0,0 +1,194 @@
+        )
+# -----------------------------------------------------------------------------------------------------
+
+    @patch.dict(
+    os.environ,
+    {
+        "DB_QUEUE_URL": "https://us-west-2.queue.amazonaws.com/123456789012/dbqueue",
+        "RECOVERY_QUEUE_URL": "https://us-west-2.queue.amazonaws.com/123456789012/recoveryqueue",
+        "MAX_RETRIES": "2",
+        "RETRY_SLEEP_SECS": "2",
+        "RETRY_BACKOFF": "2",
+    },
+    clear=True,
+    )
+    
+    def test_event_var_None_empty(self):
+        """
+        Validates that an error is thrown if an expected environment variable is not
+        set.
+        """
+        env_names = [
+            "DB_QUEUE_URL",
+            "RECOVERY_QUEUE_URL",
+            "MAX_RETRIES",
+            "RETRY_SLEEP_SECS",
+            "RETRY_BACKOFF"
+        ]
+        env_bad_values = [None, ""]
+
+        for name in env_names:
+            good_value = os.getenv(name)
+            for bad_value in env_bad_values:
+                with self.subTest(
+                    name=name, bad_value=bad_value, good_value=good_value
+                ):
+                    # Set the variable to the bad value and create the message
+                    if bad_value is None:
+                        del os.environ[name]
+                    else:
+                        os.environ[name] = bad_value
+                    # Run the test
+                    self.assertRaises(Exception, handler, self.event, context=None)
+                    # Reset the value
+                    os.environ[name] = good_value
+
+# -----------------------------------------------------------------------------------------------------
+    @patch.dict(
+        os.environ,
+        {
+            "DB_QUEUE_URL": "https://us-west-2.queue.amazonaws.com/123456789012/dbqueue",
+            "RECOVERY_QUEUE_URL": "https://us-west-2.queue.amazonaws.com/123456789012/recoveryqueue",
+            "MAX_RETRIES": "non-integer",
+            "RETRY_SLEEP_SECS": "non-integer",
+            "RETRY_BACKOFF": "non-integer",
+        },
+        clear=True,
+    )
+    def test_event_non_int(self):
+        """
+        raises ValueError when MAX_RETRIES, RETRY_SLEEP_SECS, RETRY_BACKOFF is non-integer.
+        """
+        env_names = [
+            "DB_QUEUE_URL",
+            "RECOVERY_QUEUE_URL",
+            "MAX_RETRIES",
+            "RETRY_SLEEP_SECS",
+            "RETRY_BACKOFF"
+        ]
+
+        for name in env_names:
+            value = os.getenv(name)
+            with self.subTest(
+                name=name, value=value
+            ):
+                if name in ["MAX_RETRIES", "RETRY_SLEEP_SECS", "RETRY_BACKOFF"]:
+                    self.assertRaises(ValueError, handler, self.event, context=None)
+
+    # --------------------------------------exponential delay test---------------------------------
+    def test_exponential_delay(self):
```

Comment:
Recommend including something about non-integer args in test name.

Reply:
test name changed to test_exponential_delay_non_int(self)